### ISSUE TITLE
plugin crashes on node deployment creation

### ISSUE THREAD

**Issue by bastibrunner** [syseleven/terraform-provider-metakube#42]
plugin crashes on step node deployment creation:

```
metakube_node_deployment.ops_playground_cluster1_node_deployment_1: Creating...
metakube_node_deployment.ops_playground_cluster1_node_deployment_1: Still creating... [10s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with metakube_node_deployment.ops_playground_cluster1_node_deployment_1,
│   on node-deployments.tf line 1, in resource "metakube_node_deployment" "ops_playground_cluster1_node_deployment_1":
│    1: resource "metakube_node_deployment" "ops_playground_cluster1_node_deployment_1" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-metakube_v0.5.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c7a42f]

goroutine 146 [running]:
github.com/syseleven/terraform-provider-metakube/metakube.metakubeResourceNodeDeploymentWaitForReady.func1(0x2)
        github.com/syseleven/terraform-provider-metakube/metakube/resource_metakube_node_deployment.go:376 +0x32f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.RetryContext.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/resource/wait.go:27 +0x5b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1(0xc00047c360, 0xc0004a6000, 0xc0001d4420, 0xc0001aa2c0, 0xc000376d58, 0xc000376d50)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/resource/state.go:110 +0x2e9
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/resource/state.go:83 +0x1c6

Error: The terraform-provider-metakube_v0.5.0 plugin crashed!
```

terraform -version
```
Terraform v1.0.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/syseleven/metakube v0.5.0```
```

System: MacOS Big Sur 11.6

**Comment by bastibrunner** [comment]
my node deployment config:
```
resource "metakube_node_deployment" "ops_playground_cluster1_node_deployment_1" {
  cluster_id = metakube_cluster.ops_playground_cluster1.id
  spec {
    min_replicas = 2
    max_replicas = 10
    template {
      cloud {
        openstack {
          flavor            = "m1.small"
          disk_size         = 50
          image             = "Ubuntu Focal 20.04 (2021-09-22)"
          use_floating_ip   = true
        }
      }
      operating_system {
        ubuntu {
          dist_upgrade_on_boot = true
        }
      }
    }
  }
}
```

seems to be due to autoscaling, setting static replicas value works fine

**Comment by furkhat** [comment]
@bastibrunner Thank you for opening the issue. This should be fixed with `v0.6.1`.

**Comment by bastibrunner** [comment]
can confirm, the problem is gone.
